exercises/practice/bob/.approaches/introduction.md

```diff
@@ -0,0 +1,102 @@
+
+## Approach: `if` statements
+
+```csharp
+    // uses System.Linq;
+    public static string Response(string message)
+    {
+        if (message.IsSilence())
+            return "Fine. Be that way!";
+
+        if (message.IsYell() && message.IsQuestion())
+            return "Calm down, I know what I'm doing!";
+
+        if (message.IsYell())
+            return "Whoa, chill out!";
+
+        if (message.IsQuestion())
+            return "Sure.";
+
+        return "Whatever.";
+    }
+
+    private static bool IsSilence(this string message) =>
+        string.IsNullOrWhiteSpace(message);
+
+    private static bool IsYell(this string message) =>
+        message.Any(char.IsLetter) && message.ToUpperInvariant() == message;
+
+    private static bool IsQuestion(this string message) =>
+        message.TrimEnd().EndsWith("?");
+```
+
+For more information, check the [`if` statements approach][approach-if].
+
+## Approach: `switch` on a `tuple`
+
+```csharp
+    // uses System.Linq;
+    private static bool isShout(string input) => input.Any(c => char.IsLetter(c)) && input.ToUpper() == input;
+
+    public static string Response(string statement)
+    {
+        var input = statement.TrimEnd();
+        if (input == "") return "Fine. Be that way!";
+
+        switch ((input.EndsWith('?'), isShout(input)))
+        {
+            case (true, true): return "Calm down, I know what I'm doing!";
+            case (_, true): return "Whoa, chill out!";
+            case (true, _): return "Sure.";
+            default: return "Whatever.";
+        }
+    }
+```
+
+For more information, check the [`switch` on a `tuple`][approach-switch].
+
+## Other approaches
+
+- An array can be defined that contains Bob’s answers, and each condition is given a score.
+The correct answer is selected from the array by using the score as the array index.
+For more information, check the [Answer array approach][approach-answer-array].
+
+## Which approach to use?
+
+Which to use is pretty much a matter of personal preference.
+
+Regardless of the approach used, some things to look for include
+
+- If the input is trimmed, [Trim][trim] only once.
+
+- Use the [EndsWith][endswith] `String` method instead of checking the last character by index for `?`.
+
+- Don't copy/paste the logic for determining a shout and for determing a question into determing a shouted question.
+Combine the two determinations instead of copying them.
+Not duplicating the code will keep the code [DRY][dry].
+
+- Perhaps consider making `IsQuestion` and `IsShout` values set once instead of functions that are possibly called twice.
+
+- If an `if` statement can return, then an `else if` or `else` is not needed.
+Execution will either return or will continue to the next statement anyway.
+
+- If the body of an `if` statement is only one line, curly braces aren't needed.
+Some teams may still require them in their style guidelines, though.
```

Comment:
Expression-bodied members are covered in the `if` approach content.md, since that's where they're used.

Looking ahead, I'm not sure if we want to keep repeating about single-line `if` statements and expression-bodied members. They can come up in a lot of exercises. At what point can we figure they've been exposed to certain idioms and we don't have to keep explaining them? It's easy to decide when looking at someone's individual solution whether to mention them, but for general documentation, I really don't know.

Reply:
I misread the text, as I thought it applied to methods and not `if` statement.

Reply:
> Looking ahead, I'm not sure if we want to keep repeating about single-line if statements and expression-bodied members.

We probably don't. I think it's fine if for some approaches we show a shortened expression-bodied member version and only link to the docs.

Seeing as we're talking about expression bodies a lot, let me ask you a question as a mentor: how often do you see the first iteration of a student's code using an expression body? If you think it is not _that_ uncommon, I could be swayed to start using them in our approaches. 

BTW I just noticed that the official [tour of C#](https://learn.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/program-building-blocks#members) also mentions expression bodies, so we might be fine.

Reply:
This is where it would be great to search solutions by whether they used `=>` or not.

My anecdotal mentoring impression is probably cross-contaminated by arrow functions used in other languages like JS and TS. I see them used even in TwoFer, but to what extent, sorry, I can't say offhand.

Were we going to have some kind of general document(s) for language guidelines? If we did, that would be a great place to put such things as single-line `if` statements and expression-bodied members.

Reply:
> This is where it would be great to search solutions by whether they used => or not.

Let me whip something up tomorrow. I'll get back at you.

> Were we going to have some kind of general document(s) for language guidelines?

Not sure. It might be a good idea.